java-manta-client-unshaded/src/main/java/com/joyent/manta/client/MantaClient.java

```diff
@@ -411,6 +411,341 @@ public boolean isClosed() {
+                    "Expected result-set-size header to be non-null but it was not"
+                            + " part of the response");
+            e.setContextValue("path", path);
+            throw e;
+        }
+
+        return size == 0;
+    }
+
+    /**
+     * Return a stream of the contents of a bucket in Manta.
+     *
+     * @param bucketPath The fully qualified path of a bucket.
+     * @return A {@link Stream} of {@link MantaObjectResponse} listing the contents of the bucket.
+     * @throws IOException thrown when there is a problem getting the listing over the network
+     */
+    public Stream<MantaObject> listBucketObjects(final String bucketPath) throws IOException {
+        final MantaBucketListingIterator itr = streamingBucketIterator(bucketPath);
+
+        /* We preemptively check the iterator for a next value because that will
+         * trigger an error if the path doesn't exist or is otherwise inaccessible.
+         * This error typically takes the form of an UncheckedIOException, so we
+         * unwind that exception if the cause is a MantaClientHttpResponseException
+         * and rethrow another MantaClientHttpResponseException, so that the
+         * stacktrace will point to this running method.
+         */
+        try {
+            if (!itr.hasNext()) {
+                itr.close();
+                return Stream.empty();
+            }
+        } catch (UncheckedIOException e) {
+            if (e.getCause() instanceof MantaClientHttpResponseException) {
+                throw e.getCause();
+            } else {
+                throw e;
+            }
+        }
+
+        final int additionalCharacteristics = Spliterator.CONCURRENT
+                | Spliterator.ORDERED | Spliterator.NONNULL | Spliterator.DISTINCT;
+
+        Stream<Map<String, Object>> backingStream =
+                StreamSupport.stream(Spliterators.spliteratorUnknownSize(
+                        itr, additionalCharacteristics), false);
+
+        Stream<MantaObject> stream = backingStream
+                .map(MantaObjectConversionFunction.INSTANCE)
+                .onClose(itr::close);
+
+        danglingStreams.add(stream);
+
+        return stream;
+    }
+
+
+    /**
+     * Moves a bucket from one path to another path. This operation is not
+     * transactional and may fail or produce inconsistent result if the source
+     * or the destination is modified while the operation is in progress.
+     *
+     * @param source Original path to move from
+     * @param destination Destination path to move to
+     * @param entry Bucket supplemental data object
+     * @throws IOException thrown when something goes wrong
+     * @throws MantaClientHttpResponseException Source doesn't exist or destination has invalid name
+     * @throws MantaClientException If this operation is applied to a directory.
+     */
+    private void moveBucket(final String source, final String destination,
```

Comment:
This operation seems like it has the method to be pretty dangerous, or at least require careful thought before using.  Is it an operation we need to provide in the core of the client?

Reply:
I would vote for removing this method. I'm not sure we need to support `moveBucket`.